torchvision/transforms/functional_tensor.py

```diff
@@ -48,3 +48,74 @@ def crop(img, top, left, height, width):
         raise TypeError('tensor is not a torch image.')
 
     return img[..., top:top + height, left:left + width]
+
+
+def adjust_brightness(img, brightness_factor):
+    """Adjust brightness of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        brightness_factor (float):  How much to adjust the brightness. Can be
+            any non negative number. 0 gives a black image, 1 gives the
+            original image while 2 increases the brightness by a factor of 2.
+
+    Returns:
+        Tensor: Brightness adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    black_img = torch.zeros(img.shape, device=img.device)
+
+    return _blend(img, black_img, brightness_factor)
+
+
+def adjust_contrast(img, contrast_factor):
+    """Adjust contrast of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        contrast_factor (float): How much to adjust the contrast. Can be any
+            non negative number. 0 gives a solid gray image, 1 gives the
+            original image while 2 increases the contrast by a factor of 2.
+
+    Returns:
+        Tensor: Contrast adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    mean = torch.mean(_rgb_to_grayscale(img))
+    mean_img = mean * torch.ones(img.shape, device=img.device)
+
+    return _blend(img, mean_img, contrast_factor)
+
+
+def adjust_saturation(img, saturation_factor):
+    """Adjust color saturation of an RGB image.
+
+    Args:
+        img (Tensor): Image to be adjusted.
+        saturation_factor (float):  How much to adjust the saturation. 0 will
+            give a black and white image, 1 will give the original image while
+            2 will enhance the saturation by a factor of 2.
+
+    Returns:
+        Tensor: Saturation adjusted image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    num_channels = img.shape[0]
+    gray_img = _rgb_to_grayscale(img).repeat(num_channels, 1, 1)
+
+    return _blend(img, gray_img, saturation_factor)
+
+
+def _blend(img1, img2, ratio):
+    return (ratio * img1 + (1 - ratio) * img2).clamp(0, 1)
```

Comment:
This is great, thanks!

I was wondering if should make `_blend` return an output of the same type as the input. We now have automatic type promotion in PyTorch, so `uint8 * float` becomes a `float`. This means that if the user passed an `uint8` image, he would have a float tensor back.

If we want to support this use-case (which could be nice), we would need to pay a bit more attention with the clamping boundaries.
One possibility would be to use `torch.iinfo` if the input tensor is of an integer type (via `img1.dtype.is_floating_point`) to get the min / max bounds for clamp, and set it to 0-1 if the input is floating point.

Thoughts?

Reply:
Done. I'm assuming here that the upper bound for an int image is always 255, not sure if we support images with a bigger range.